### PR TITLE
Replace Google OAuth with database authentication

### PR DESCRIPTION
**Pull request by DuBistKomisch**
Fixes https://github.com/noutube/issues/issues/1

### CODE REVIEW

File: app/pods/application/route.ts [hunk 1]
```diff
@@ -1,11 +1,13 @@
 import Transition from '@ember/routing/-private/transition';
 import Route from '@ember/routing/route';
+import RouterService from '@ember/routing/router-service';
```
Comment (DuBistKomisch) [inline app/pods/application/route.ts:3]: these are leftovers

File: app/pods/account/route.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  @service declare router: RouterService;
+  @service declare session: SessionService;
+
+  async beforeModel(transition: Transition): Promise<void> {
+    if (!this.session.me && !this.session.down) {
+      this.router.transitionTo('landing');
+    }
+    if (this.session.me) {
+      this.session.me.rollbackAttributes();
+    }
```
Comment (deanylev) [inline app/pods/account/route.ts:18]: Could nest `this.session.down` instead

File: app/pods/components/route-register/template.hbs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  @icon="user-plus"
+>
+  Register
+</SectionHeader>
+
+<div local-class="register">
+  <div local-class="row">
+    <label>
+      Email
+      <Input
```
Comment (deanylev) [inline app/pods/components/route-register/template.hbs:13]: You might as well use a normal HTML input if you're not relying on two way bindings. Isn't passing `this.email` as the value just going to be setting that twice?

File: app/pods/components/route-register/template.hbs [hunk 4]
```diff
@@ -0,0 +1,53 @@
+<SectionHeader
+  @backIcon="home"
+  @backRoute="landing"
+  @icon="user-plus"
+>
+  Register
+</SectionHeader>
+
+<div local-class="register">
```
Comment (deanylev) [inline app/pods/components/route-register/template.hbs:9]: Surely this an `route-sign-in` can be a common component? They're 95% identical

File: app/pods/components/section-header/styles.scss [hunk 5]
```diff
@@ -7,12 +7,22 @@
...
-  padding: 8px 16px;
+  padding: 0 16px 0 8px;
   min-height: 24px;
   font-size: 125%;
   font-weight: bold;
 
+  > .chevron {
+    @include theme((color: bg-hi));
+
+    margin: 0 -4px;
```
Comment (deanylev) [inline app/pods/components/section-header/styles.scss:18]: Negative margin :monkachrist:

**Pull request merged**